src/deluge/model/instrument/non_audio_instrument.cpp

```diff
@@ -96,11 +116,24 @@ void NonAudioInstrument::sendNote(ModelStackWithThreeMainThings* modelStack, boo
 		// Run everything by the Arp...
 		arpeggiator.noteOff(arpSettings, noteCodePreArp, &instruction);
 
-		if (instruction.noteCodeOffPostArp != ARP_NOTE_NONE) {
-			noteOffPostArp(instruction.noteCodeOffPostArp, instruction.outputMIDIChannelOff, velocity);
+		for (int32_t n = 0; n < ARP_MAX_INSTRUCTION_NOTES; n++) {
+			if (instruction.noteCodeOffPostArp[n] != ARP_NOTE_NONE) {
+				noteOffPostArp(instruction.noteCodeOffPostArp[n], instruction.outputMIDIChannelOff[n], velocity, n);
+			}
+			else {
+				break;
+			}
 		}
-		if (instruction.noteCodeOnPostArp != ARP_NOTE_NONE && type == OutputType::CV) {
-			noteOnPostArp(instruction.noteCodeOnPostArp, instruction.arpNoteOn);
+		// CV instruments could switch on a note to do a glide
+		if (type == OutputType::CV) {
+			for (int32_t n = 0; n < ARP_MAX_INSTRUCTION_NOTES; n++) {
+				if (instruction.arpNoteOn != nullptr && instruction.arpNoteOn->noteCodeOnPostArp[n] != ARP_NOTE_NONE) {
```

Comment:
Suggest same transform as above (move `instruction.arpNoteOn` nullptr check out of the loop and make `ARP_NOTE_NONE` check a filter).